core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -960,19 +960,27 @@ private[spark] class TaskSetManager(
       assert (null != failureReason)
       taskSetExcludelistHelperOpt.foreach(_.updateExcludedForFailedTask(
         info.host, info.executorId, index, failureReason))
-      numFailures(index) += 1
-      if (numFailures(index) >= maxTaskFailures) {
-        logError("Task %d in stage %s failed %d times; aborting job".format(
-          index, taskSet.id, maxTaskFailures))
-        abort("Task %d in stage %s failed %d times, most recent failure: %s\nDriver stacktrace:"
-          .format(index, taskSet.id, maxTaskFailures, failureReason), failureException)
-        return
+      if (!successful(index)) {
+        numFailures(index) += 1
+        if (numFailures(index) >= maxTaskFailures && !successful(index)) {
+          logError("Task %d in stage %s failed %d times; aborting job".format(
+            index, taskSet.id, maxTaskFailures))
+          abort("Task %d in stage %s failed %d times, most recent failure: %s\nDriver stacktrace:"
+            .format(index, taskSet.id, maxTaskFailures, failureReason), failureException)
+          return
+        }
       }
     }
 
     if (successful(index)) {
-      logInfo(s"${taskName(info.taskId)} failed, but the task will not" +
-        " be re-executed (either because the task failed with a shuffle data fetch failure," +
+      val message = if (numFailures(index) >= maxTaskFailures) {
```

Comment:
Why should we update this log message?

Reply:
This log is added according to the advice from @Ngone51, maybe we need add some log to notice user for this case.

Reply:
It seems if we reset the `numFailures`, this log should be rollback, i will rollback this log